example_code/applications/crowd_fund/src/lib.rs

```diff
@@ -0,0 +1,203 @@
+#![cfg_attr(not(feature = "export-abi"), no_main)]
+extern crate alloc;
+use alloy_sol_types::sol;
+
+use stylus_sdk::{
+    alloy_primitives::{Address, U256},
+    block,
+    call::Call,
+    contract, evm, msg,
+    prelude::*,
+};
+
+sol_interface! {
+    interface IERC20 {
+    function transfer(address, uint256) external returns (bool);
+    function transferFrom(address, address, uint256) external returns (bool);
+    }
+}
+
+sol! {
+    event Launch(
+        uint256 id,
+        address indexed creator,
+        uint256 goal,
+        uint256 start_at,
+        uint256 end_at
+    );
+    event Cancel(uint256 id);
+    event Pledge(uint256 indexed id, address indexed caller, uint256 amount);
+    event Unpledge(uint256 indexed id, address indexed caller, uint256 amount);
+    event Claim(uint256 id);
+    event Refund(uint256 id, address indexed caller, uint256 amount);
+}
+
+sol_storage! {
+    #[entrypoint]
+    pub struct CrowdFund{
+    // Total count of campaigns created.
+    // It is also used to generate id for new campaigns.
+    uint256 count;
+    // The address of the NFT contract.
+    address token_address;
+    // Mapping from id to Campaign
+    CampaignStruct[] campaigns; // The transactions array
+    // Mapping from campaign id => pledger => amount pledged
+    mapping(uint256 => mapping(address => uint256)) pledged_amount;
+    }
+    pub struct CampaignStruct {
+        // Creator of campaign
+        address creator;
+        // Amount of tokens to raise
+        uint256 goal;
+        // Total amount pledged
+        uint256 pledged;
+        // Timestamp of start of campaign
+        uint256 start_at;
+        // Timestamp of end of campaign
+        uint256 end_at;
+        // True if goal was reached and creator has claimed the tokens.
+        bool claimed;
+    }
+
+}
+
+/// Declare that `CrowdFund` is a contract with the following external methods.
+#[public]
+impl CrowdFund {
+    pub const ONE_DAY: u64 = 86400; // 1 day = 24 hours * 60 minutes * 60 seconds = 86400 seconds.
+
+    pub fn launch(&mut self, goal: U256, start_at: U256, end_at: U256) {
+        assert!(start_at < U256::from(block::timestamp()));
+        assert!(end_at < start_at);
+        assert!(end_at > U256::from(block::timestamp() + 7 * Self::ONE_DAY));
```

Comment:
These Rust-native assertions add to code bloat, iirc. I don't think we should be recommending their usage except in off-chain testing scripts. Just use `if` or `match` statements. (you can also try commenting out all assertions to see if it makes an appreciable difference in code size / execution gas)